### ISSUE TITLE
[Sourcing-2][Accessibility: ACC-264.1][WIP] This DIV is focusable and has an aria-hidden attribute set to true.

### ISSUE THREAD

**Issue by JyotiKumari1507** [SAP/fundamental-ngx#11278]
#### Is this a bug, enhancement, or feature request?
Bug

#### Describe your proposal.
Component having issue: https://sap.github.io/fundamental-ngx/#/platform/table/basic
select any row from the table:
This DIV is focusable and has an aria-hidden attribute set to true.

Code snippet: <div aria-hidden="true" class="cod-visually-hidden cdk-focus-trap-anchor" tabindex="0"></div>

Content providers should generally ensure that only active elements receive keyboard focus. Non-interactive elements such as labels and headings should be removed from the focus order by not having a tabindex attribute set. For interactive elements that are in the focus order but rendered inactive, tabindex="-1" can be used to remove them from the focus order. In certain situations such as controls like submit buttons that are currently disabled until the user enters the correct information it may be helpful to have the control in the focus order with a message letting the user know what must be completed in order to have it become enabled. In these cases proper markup such as aria-disabled may be needed to communicate the disabled state of the control to the user of assistive technology. Another example of inactive content that should be keyboard focusable is read-only content that must be copied such as codes, etc.

Setting focus to non-interactive elements programmatically such as after a form is submitted to focus an error message can also be an acceptable technique. Generally focus should not be forcibly shifted this way -- but in some rare situations it is acceptable. Another example of a rare situation is setting focus to a time out dialog.

When SVG elements are used some user agents such as Internet Explorer will place these elements in the focus order unless the focusable property is explicitly set to false. Thus, if the SVG is not interactive the focusable property should not be set to true and ideally would explicitly be set to false.

#### Can you handle that on the application side
No

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)
All latest version

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;
•	Download Access assistant tool from level access.
•	For registered AMP user, login with your SAP email ID into SAP AMP instance: sap.levelaccess.net
select any row from the table:
This DIV is focusable and has an aria-hidden attribute set to true.

#### In case this is Accessibility related topic, did you consult with an accessibility expert? If not, please do so and share their recommendations. 
Yes we have got this bug from Accessibility expert only.

#### Did you check the documentation and the API?
Yes

#### Is there anything else we should know?

**Comment by nikolay-kolarov** [comment]
HI @JyotiKumari1507,

Could you please provide more info or a screenshot of which table in the test page happens this?

Thanks,
Nikolay

**Comment by JyotiKumari1507** [comment]
@nikolay-kolarov  Please find attached screenshot shared by accessibility team

**Comment by nikolay-kolarov** [comment]
Hello @droshev,

The problem is not related with the table, but it comes from the dialog and is reproducible on the dialog page (when you open a dialog):
https://sap.github.io/fundamental-ngx/#/core/dialog
The complain comes from the two elements used to trap the focus in the dialog: 
![image](https://github.com/SAP/fundamental-ngx/assets/17043004/dace7ecd-2aea-42f3-a97f-27b563a545e8)
Due to the tabindex, the validator assumes that those are really focusable and complains that they have aria hidden in the same time.

As a fix, you can get the solution from the UI5 dialog: https://ui5.sap.com/#/entity/sap.m.Dialog/sample/sap.m.sample.Dialog
![image](https://github.com/SAP/fundamental-ngx/assets/17043004/1eb4bbf2-23b8-46ee-abb5-d971e5d78830)

Best Regards,
Nikolay

**Comment by mikerodonnell89** [comment]
This is not coming from fundamental-ngx but from Angular, and appears to be deliberate. See https://github.com/angular/components/blob/main/src/cdk/a11y/focus-trap/focus-trap.ts#L332 and https://github.com/angular/components/blob/main/src/cdk/a11y/focus-trap/focus-trap.ts#L344

Closing this issue here
